apps/organisations/views.py

```diff
@@ -215,44 +218,93 @@ def get_context_data(self, **kwargs):
 
     def form_valid(self, form):
         data = form.cleaned_data
-        img = self.generate_image(data)
         context = self.get_context_data()
+        img = self.generate_image(data)
+        # get the content form with unchanged cleaned data, as image generation
+        # changes the image data (why?)
+        content_form = CommunicationContentCreationForm(
+            initial=data, project=self.project, format=self.format)
         context['image_preview'] = img
+        context['content_form'] = content_form
         return self.render_to_response(context)
 
     def generate_image(self, data):
-        # Dummy images
-        image = Image.new('RGB', (304, 192), color='red')
-        logo1 = Image.new('RGB', (100, 10), color='blue')
-        logo2 = Image.new('RGB', (100, 10), color='green')
-
-        # Adding padding
+        # Get required image size and image
+        sharepic_format = SOCIAL_MEDIA_SIZES[self.format]
+        image_get = Image.open(data['image'].file)
+        image_size = (sharepic_format['img_min_width'],
+                      sharepic_format['img_min_height'])
+        image = image_get.resize(image_size)
```

Comment:
If the a+ logo is not included there is a lot of weird looking white padding at the bottom. One way to prevent this could be to increase the resize height here to something like `img_min_height` + `size of a+ logo`